### ISSUE TITLE
Feature - in v 3.0.2 needs a way to choose directory for saving project (on windows) and overwriting existing file

### ISSUE THREAD

**Issue by vgreff** [sharpie7/circuitjs1#975]
The latest version now  just saves in the windows download directory without asking. It no longer opens a dialog box to change the directory to save the file and it also keeps creating new versions of file for every save (same name and adds (##) suffix. 
There is no way to replace/overwrite the previous file.
A menu option to choose the working directory would be nice and an option to allow overwriting an existing file

**Comment by pfalstad** [comment]
Hmm, that's odd because I didn't change anything related to this.  I just changed the filename format.  I'm also not sure if I have any control over whether the browser decides to display a dialog box.

This is the online version, right? What browser is it, Chrome?

**Comment by vgreff** [comment]
Thank you for the quick answer and 

Yes it’s in the online version 3.0.2 on Falstad.com (old version 2.9.1 brings up a dialog box) and I’m using firefox (v133) on windows 10/11.

In 3.0.2 the save is kind of silent after the first dialog box asking the file name (no question asked via further dialog box), on 2.9.6 CRTL-s (save) brings up a second dialog box afterans is  asking for the directory where to save and a third dialog box if there is a conflict with existing file name.

It’s a bit too many dialog boxes IMHO, but woks.

 

One suggestion would be add a field for the directory name location in the very first dialog box that asks for the name (it’s clearly picking a directory but the user does not know where it is nor can’t change it) and maybe a checkbox that says overwrite file if exists

 

BTW, I love the app 😊

**Comment by pfalstad** [comment]
I tried firefox on windows and it acts the same with the old version.

You can try 2.9.0 here: https://www.falstad.com/circuit/circuitjs96.html

It's the browser that's asking what directory to save in.  The web app has no control over that.  It can't check to see if there's a file there already either.  For security reasons, the web app can't look at your files.  I think there might be a new extension which allows that but I don't think firefox supports it anyway.

Maybe there is a browser setting that controls that, which got changed recently?

**Comment by vgreff** [comment]
Problem fixed, I can confirm now and  that the behavior in 3.0.3 is the same to 2.9.1. Dialog boxes are all back.

Polarized caps fixed worked too BTW as you said in other mail

 

I had to manually clear 7.2MB of data that associated with Falstad.com in the firefox browser on my windows 10 machine and then the 3.0.3 version showed up there as well. Without manually clearing it did not auto upgrade to 3.0.3, not a big deal but I wanted to share that data point.

 

Vincent B. Greff

**Comment by pfalstad** [comment]
Yeah I noticed there's something going on with the caching, I need to figure that out
